sympy/solvers/solveset.py

```diff
@@ -76,6 +76,14 @@ def _invert_real(f, g_ys, symbol):
                             imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, Abs):
+        _a = []
+        for i in g_ys:
+            try:
+                if i < 0:
```

Comment:
After discussion on #9586 i think,  we should do it the way it is done in second commit. Since for after my second commit, this is the thing
```
>>> x = Symbol('x', real=True)
>>> y = Symbol('y')
>>> invert_real(Abs(x) + 1, y, x)
(x, {-y + 1, y - 1})
```